### PR TITLE
PHYVersion API clarifications

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

References https://github.com/TheThingsNetwork/lorawan-stack/issues/3513

#### Changes

- Rename the fields of the `PHYVersion` enum based on the details enumerated here https://github.com/TheThingsNetwork/lorawan-stack/issues/2784#issuecomment-648735723 . The renamed enum values are backwards compatible in their JSON representation and binary representation.
- Add the RP2 `PHYVersion`s.
- Clarify in `pkg/band` which RP/TS versions are used for different versions of the band.
- Changes usages of the old naming scheme to the new naming scheme.
- Rename `MinAckTimeout` and `MaxAckTimeout` to `MinRetransmitTimeout` and `MaxRetransmitTimeout`, as this is the terminology used by RP2.

#### Testing

Unit testing.

##### Regressions

There should be none, as most of the work is just automated renaming.

#### Notes for Reviewers

This PR is meant to be the initial work for https://github.com/TheThingsNetwork/lorawan-stack/issues/3513 . It allows us to have correct names for the regional parameters without breaking compatibility.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/devicerepository/store/remote/enums.go [hunk 1]
```diff
@@ -38,16 +38,16 @@ var bandIDToRegion map[string]string
...
-	// "RP002-1.0.0": ttnpb.PHY_UNKNOWN,
-	// "RP002-1.0.1": ttnpb.PHY_UNKNOWN,
-	"RP001-1.1":      ttnpb.PHY_V1_1_REV_A,
-	"RP001-1.1-RevB": ttnpb.PHY_V1_1_REV_B,
+	"TS001-1.0":        ttnpb.TS001_V1_0,
+	"TS001-1.0.1":      ttnpb.TS001_V1_0_1,
+	"RP001-1.0.2":      ttnpb.RP001_V1_0_2,
+	"RP001-1.0.2-RevB": ttnpb.RP001_V1_0_2_REV_B,
+	"RP001-1.0.3-RevA": ttnpb.RP001_V1_0_3_REV_A,
+	"RP001-1.1":        ttnpb.RP001_V1_1,
```
Comment (adriansmares) [inline pkg/devicerepository/store/remote/enums.go]: This list has been re-ordered based on the TS001/RP001/RP002 releases.

File: pkg/networkserver/networkserver_flow_test.go [hunk 2]
```diff
@@ -60,7 +60,18 @@ func frequencyPlanMACCommands(macVersion ttnpb.MACVersion, phyVersion ttnpb.PHYV
 				NbTrans:       1,
 			},
 		}
-		if !otaa || phyVersion.Compare(ttnpb.PHY_V1_0_3_REV_A) < 0 {
+		beforeRP001_V1_0_3_REV_A := func(v ttnpb.PHYVersion) bool {
```
Comment (adriansmares) [inline pkg/networkserver/networkserver_flow_test.go:63]: I've removed the ordering of versions, as `RP002` has clashing definitions for `1.0.0` and `1.0.1`.

File: pkg/ttnpb/testdata/ttnpb_encoding_golden.md [hunk 3]
```diff
@@ -594,6 +597,9 @@ JSON | ttnpb.PHYVersion | 1.0.2-b | "PHY_V1_0_2_REV_B"
 JSON | ttnpb.PHYVersion | 1.0.3-a | "PHY_V1_0_3_REV_A"
 JSON | ttnpb.PHYVersion | 1.1.0-a | "PHY_V1_1_REV_A"
 JSON | ttnpb.PHYVersion | 1.1.0-b | "PHY_V1_1_REV_B"
+JSON | ttnpb.PHYVersion | RP002_V1_0_0 | "RP002_V1_0_0"
```
Comment (adriansmares) [inline pkg/ttnpb/testdata/ttnpb_encoding_golden.md:601]: Note that the old representations are unchanged.

File: pkg/ttnpb/testdata/ttnpb_encoding_golden.md [hunk 4]
```diff
@@ -1182,6 +1188,9 @@ Text | ttnpb.PHYVersion | 1.0.2-b | 1.0.2-b
 Text | ttnpb.PHYVersion | 1.0.3-a | 1.0.3-a
 Text | ttnpb.PHYVersion | 1.1.0-a | 1.1.0-a
 Text | ttnpb.PHYVersion | 1.1.0-b | 1.1.0-b
+Text | ttnpb.PHYVersion | RP002_V1_0_0 | RP002_V1_0_0
```
Comment (adriansmares) [inline pkg/ttnpb/testdata/ttnpb_encoding_golden.md:1193]: Note that the old representations are unchanged.

**Review by htdvisser — APPROVED** [review]

File: api/lorawan.proto [hunk 5]
```diff
@@ -81,13 +81,19 @@ enum PHYVersion {
...
-  PHY_V1_1_REV_A = 5;
-  PHY_V1_1_REV_B = 6;
-  PHY_V1_0_3_REV_A = 7;
+
+  TS001_V1_0 = 1;
+  TS001_V1_0_1 = 2;
+
+  RP001_V1_0_2 = 3;
+  RP001_V1_0_2_REV_B = 4;
+  RP001_V1_1 = 5;
```
Comment (johanstokking) [inline api/lorawan.proto]: This one is actually Rev A

**Review by johanstokking — COMMENTED** [review]
I think you should take a shower after merging this PR

**Review by KrishnaIyer — APPROVED** [review]
May god be with you.

**Comment by johanstokking** [comment]
@adriansmares are you holding the merge for the decisions and implications regarding PHY? Or what is the reason to hold?

**Comment by adriansmares** [comment]
> 
> 
> @adriansmares are you holding the merge for the decisions and implications regarding PHY? Or what is the reason to hold?

It needs approval from a code owner - you and/or @neoaggelos .

**Review by johanstokking — APPROVED** [review]

**Review by neoaggelos — APPROVED** [review]
Also tested backwards compatibility with previous enum values (e.g. `PHY_V1_0_2_REV_B`) to make sure that nothing breaks. :+1:

**Pull request merged**